### PR TITLE
STYLE: Improve readability of HexadedronCell::EvaluatePosition.

### PR DESCRIPTION
**Pull request by hjmjohnson**
3 is the cell dimension -- make this explicit.

Change-Id: I7f458d7517ddb0442c05a35dde1af8734f8786f4

### CODE REVIEW

**Comment by hjmjohnson** [comment]
Matt McCormick
Uploaded patch set 1.Oct 15, 2015

Hans J. Johnson
Patch Set 1: (2 comments)Oct 15, 2015

Matt McCormick
Patch Set 1: (1 comment)Oct 15, 2015

Matt McCormick
Uploaded patch set 2.Oct 15, 2015

Kitware Build Robot
Patch Set 1: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=20279-1&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 15, 2015

Gert Wollny
Patch Set 2: regarding the PointDimensions it seems fine, but the 8 should be Self::NumberOfVertices or Self:: NumberOfPoints instead. No need to add another constant.Oct 15, 2015

Hans J. Johnson
Patch Set 2: I agree with Gert. Also, I think Self::NumberOfVertices could be used more agressively throughout this entire file. HansOct 15, 2015

Kitware Build Robot
Patch Set 2: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=20279-2&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 15, 2015

Arnaud Gelas
Patch Set 2: (1 comment)Oct 15, 2015

Gert Wollny
Patch Set 2: Arnaud is right, and based on this the class should probably be called CuboidCell ...Oct 15, 2015

Matt McCormick
Patch Set 2: OK, using Self::CellDimension and Self::NumberOfPoints. Also adding a note in the doxygen documentation that it presumes a cuboid.Oct 15, 2015

Matt McCormick
Uploaded patch set 3.Oct 15, 2015

Kitware Build Robot
Patch Set 3: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=20279-3&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 15, 2015

Gert Wollny
Patch Set 3: If you do it like this, then there should at least be some assertion that Self::CellDimension and PointDimension are the same, because now, using 2D points results in undefined behaviour and with 4+D points extra values are silently discarded. Given that the type of data that might be handled seems to depend on user input itkAssertInDebugOrThrowInReleaseMacro may be a good option.Oct 15, 2015

Matt McCormick
Patch Set 3: As remarked in ITK-3386, it may be possible where HexadedronCell is used in a context where PointDimension is 4. But, I will leave checks for valid dimension correspondence to another patch. request build: allOct 15, 2015

Kitware Build Robot
Patch Set 3: Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=20279-3&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 15, 2015

Arnaud Gelas
Patch Set 3: Code-Review-1 (1 comment)Oct 16, 2015

Matt McCormick
Patch Set 3: (1 comment)Oct 16, 2015

Matt McCormick
Uploaded patch set 4.Oct 16, 2015

Kitware Build Robot
Patch Set 4: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=20279-4&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 16, 2015

Matt McCormick
Uploaded patch set 5.Oct 16, 2015

Kitware Build Robot
Patch Set 5: Verified-1 Build Failed: CDash filtered results: https://open.cdash.org/index.php?&project=Insight&filtercount=3&field1=buildname/string&compare1=63&value1=20279-5&field2=buildstarttime/date&compare2=83&value2=2015-03-01&field3=buildstarttime/date&compare3=84&value3=2029-01-01Oct 16, 2015

Arnaud Gelas
Patch Set 5: Matt, If you really want to fix this code, you should, first of all, write tests in dimension=3 and one in dimension=4... Looking at the code, it seems to me that it did not work in dimension=4 (it was only done for dimension=3), and this patch won't fix anything...Oct 19, 2015

Hans J. Johnson
Uploaded patch set 6: Patch Set 5 was rebased.Mar 31, 2016

Kitware Build Robot

**Comment by hjmjohnson** [comment]
@thewtex This was your patch originally.  I'd prefer if you did the rebase/merge.

**Comment by thewtex** [comment]
I will take a look at some point. It seems I should follow-up with @arnaudgelas 's comment

> If you really want to fix this code, you should, first of all, write tests in dimension=3 and one in dimension=4... Looking at the code, it seems to me that it did not work in dimension=4 (it was only done for dimension=3), and this patch won't fix anything...Oct 19, 2015

**Comment by hjmjohnson** [comment]
@thewtex Perhaps make an issue to address later about adding more tests, and get this incremental small improvement pushed through?

**Comment by hjmjohnson** [comment]
@thewtex Otherwise we should make this a WIP.  so that we can ignore it for a while.

File: Modules/Core/Common/include/itkHexahedronCell.h [hunk 1]
```diff
@@ -123,10 +123,14 @@ class ITK_TEMPLATE_EXPORT HexahedronCell:public TCellInterface, private Hexahedr
   /** Store the number of points needed for a hexahedron. */
   PointIdentifier m_PointIds[NumberOfPoints];
 
-  void InterpolationDerivs(CoordRepType pcoords[3], CoordRepType derivs[24]);
-  void InterpolationFunctions(CoordRepType pcoords[3], InterpolationWeightType sf[8]);
-  void EvaluateLocation(int &itkNotUsed(subId), PointsContainer * points, CoordRepType pcoords[3],
-                        CoordRepType x[3], InterpolationWeightType * weights);
+  void
+  InterpolationDerivs( CoordRepType pcoords[Self::CellDimension],
```
[the remark this answers is not in the document]
Comment (thewtex) [reply Modules/Core/Common/include/itkHexahedronCell.h]: That sounds better, but we should retain for the current name for backwards compatibility.

**Comment by thewtex** [comment]
@hjmjohnson good point -- this was just addressing readability. I created an issue in #265 to add new tests. I will rebase...

**Comment by hjmjohnson** [comment]
@thewtex If you are rebasing, dzenan had a very minor variable name request unrelated to the changes in the codebase. I would only do that if it is really easy.

File: Modules/Core/Common/include/itkHexahedronCell.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Core/Common/include/itkHexahedronCell.h]: Ooops .  I did not refresh my browser  You comment is spot on here.

**Pull request merged**